### DISCUSSION TITLE
Headers and footers with landscape pages

### DISCUSSION THREAD

**Discussion by carlosvergara** [davidgohel/officedown#103] [Q&A] [answered]
Hi,
I am trying to knit a document using a template that includes a header and page numbering, but when using landscape pages all the pages preceding them lose the header and footer (maybe something related to #67).
Is there any way to apply header and footer to the whole document?
Thanks in advance! :)
Template: [reference.docx](https://github.com/davidgohel/officedown/files/10676723/reference.docx)

````
---
date: "`r Sys.Date()`"
author: "Your Name"
title: "officedown template"
output: 
  officedown::rdocx_document:
    reference_docx: reference.docx
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.cap = TRUE)
library(officedown)
library(officer)
```
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla eu pulvinar arcu, quis aliquam dui. In at cursus ante. Vestibulum non sagittis lacus. Duis vitae iaculis dui. Vivamus tempor, nibh ut pretium tempus, enim lorem dignissim quam, at euismod massa magna at magna. Sed facilisis dapibus diam nec volutpat. 

<!---BLOCK_LANDSCAPE_START--->

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla eu pulvinar arcu, quis aliquam dui. In at cursus ante. Vestibulum non sagittis lacus. Duis vitae iaculis dui. Vivamus tempor, nibh ut pretium tempus, enim lorem dignissim quam, at euismod massa magna at magna. Sed facilisis dapibus diam nec volutpat. 

```{r, echo=FALSE}
barplot(1:5, col=1:5)
```
<!---BLOCK_LANDSCAPE_STOP--->

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla eu pulvinar arcu, quis aliquam dui. In at cursus ante. Vestibulum non sagittis lacus. Duis vitae iaculis dui. Vivamus tempor, nibh ut pretium tempus, enim lorem dignissim quam, at euismod massa magna at magna. Sed facilisis dapibus diam nec volutpat. 

\pagebreak

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla eu pulvinar arcu, quis aliquam dui. In at cursus ante. Vestibulum non sagittis lacus. Duis vitae iaculis dui. Vivamus tempor, nibh ut pretium tempus, enim lorem dignissim quam, at euismod massa magna at magna. Sed facilisis dapibus diam nec volutpat. 

````

**Comment by davidgohel** [comment, marked as answer]
> Is there any way to apply header and footer to the whole document?

Yes. If you add a new section, you should add also the header and footer you want them to contain. Headers and footers are part of sections. 

Since few versions, it is possible to add any content in your header's footer or header. See https://davidgohel.github.io/officer/reference/prop_section.html

**Reply by carlosvergara** [reply]
David, thank you for your response and your work! : )

**Reply by davidgohel** [reply]
Glad it helped :)

**Reply by PatrickRWright** [reply]
@carlosvergara any chance of posting what exactly you added to your example document after receiving this answer here?

**Reply by davidgohel** [reply]
@PatrickRWright did you try copy and paste the examples provided in the manual? https://davidgohel.github.io/officer/reference/prop_section.html#ref-examples

**Reply by PatrickRWright** [reply]
Yes I tried. I think my problem is "marrying" `officer` code with an `officedown` document.

**Reply by davidgohel** [reply]
In the answer of your question here https://github.com/davidgohel/officedown/discussions/138#discussioncomment-8940713, there is a link that show how to "marry" them (use `block_section()` as explained, you can copy and paste other examples here https://davidgohel.github.io/officer/reference/prop_section.html#ref-examples). Can you confirm you see the answer?

**Reply by PatrickRWright** [reply]
From there I have figured out that I can do something like in the snippet below. This is, however, overriding the setting from the template that I would be using. For the sample case below that means that the page numbers do not start until page three while I would like them to start on page one.

````
---
output: officedown::rdocx_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(officedown) # 0.3.0
library(officer)
```

<!---BLOCK_TOC--->

```{r}
header_default <- block_list(fpar(ftext("text for default page header")))
footer_default <- block_list(fpar(ftext("text for default page footer")))

block_section(prop_section(header_default = header_default,
                           footer_default = footer_default))
```

# Chapter 1
# Chapter 2
<!---BLOCK_LANDSCAPE_START--->
# Chapter 3
<!---BLOCK_LANDSCAPE_STOP{break_page: "nextPage"}--->
# Chapter 4
\newpage
# Chapter 5
\newpage
# Chapter 6
````

**Reply by davidgohel** [reply]
Then add the page number in the sections you have created. I think it is explained in the documentation??!! Read, https://ardata-fr.github.io/officeverse/paragraphs-chunks.html, open the manual of `run_word_field()`

**Reply by PatrickRWright** [reply]
Ok fair enough. Thanks again!

For the record, this appears to be the actual code when just considering the page number in the footer:  
````
---
output: officedown::rdocx_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(officedown) # 0.3.0
library(officer) # 0.6.2
```

<!---BLOCK_TOC--->

```{r}
footer_default <- block_list(fpar(run_word_field(field = "PAGE"),
                                  fp_p = fp_par(text.align = "center") ))

block_section(prop_section(footer_default = footer_default))
```

# Chapter 1
# Chapter 2
<!---BLOCK_LANDSCAPE_START--->
# Chapter 3
<!---BLOCK_LANDSCAPE_STOP{break_page: "nextPage"}--->
# Chapter 4
\newpage
# Chapter 5
\newpage
# Chapter 6
````

**Comment by SnokeOne** [comment]
I having similar issue, lossing all layout borders preceding block section. Is there a way to add borders to the section?
